spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestViews.java

```diff
@@ -350,39 +350,21 @@ public void readFromViewReferencingAnotherView() throws NoSuchTableException {
   }
 
   @Test
-  public void readFromViewReferencingTempView() throws NoSuchTableException {
```

Comment:
I think the original version of this test is still valid. We want to fail if the view references a temp view because the temp view will not be resolved. The fact that we can't create a view that references a temp view in SQL is a separate consideration to test.

This is a good demonstration of why it's good to test with both SQL view creation and API view creation.

Reply:
yeah I wasn't sure exactly how we'd want to handle these cases. I agree that those are separate concerns, because in one case it should fail when the view is being created and in the other case it should fail when the view is being read from. I'll add a separate test case for this